lyrics-fetcher-neteasecloud.el

```diff
@@ -0,0 +1,216 @@
+after the request.
+
+When EDIT is non-nil, edit the query in minibuffer before search."
+  (message "Sending a query to NeteaseCloud API...")
+  (request "http://music.163.com/api/search/get/"
+    :type "POST"
+    :data `(("s" . ,(lyrics-fetcher-neteasecloud--maybe-edit-query
+		     (lyrics-fetcher-neteasecloud--format-query track)
+		     edit))
+	    ("limit" . "10")
+	    ("type" . "1")
+	    ("offset" . "0"))
+    :parser 'json-read
+    :sync sync
+    :success (cl-function
+              (lambda (&key data &allow-other-keys)
+                (funcall callback data)))
+    :error (cl-function
+            (lambda (&key error-thrown &allow-other-keys)
+              (message "Error!: %S" error-thrown)))
+    ))
+
+(defun lyrics-fetcher-neteasecloud--maybe-edit-query (query edit)
+  "If EDIT is non-nil, edit QUERY in minibuffer."
+  (if edit
+      (read-from-minibuffer "Query: " query)
+    query))
+
+(defun lyrics-fetcher-neteasecloud--format-query (track)
+  "Format track to 'music.163.com' query.
+
+When `lyrics-fetcher-neteasecloud-strip-parens-from-query' is non-nil,
+remove all the text in parens from the query,
+for instance (feat.  someone).
+
+TRACK should either be a string or an EMMS-compatible alist, which
+contains `info-artist' or `info-title'"
+  (if (stringp track)
+      track
+    (let ((query (concat
+		  (cdr (assoc 'info-title track))
+                  " "
+		  (cdr (assoc 'info-artist track)))))
+      (when lyrics-fetcher-neteasecloud-strip-parens-from-query
+        (setq query (replace-regexp-in-string
+                     (rx (or (: "(" (* nonl) ")")
+                             (: "[" (* nonl) "]")))
+                     "" query)))
+      query)))
+
+(defun lyrics-fetcher-neteasecloud--get-song-id (data &optional ask)
+  "Retrieve a song id from the 'music.163.com' response DATA.
+
+If ASK is non-nil, prompt the user for a choice, otherwise select the
+first song."
+  (if (/= 200 (alist-get 'code data))
+      (error "ERROR: %s" (alist-get 'code data))
+    (let* ((results (alist-get 'songs (alist-get 'result data))))
+      (if (seq-empty-p results)
+	  (error "ERROR: no results!")
+	(cdr
+	 (if ask
+	     (let ((results-songs-for-select
+		    (mapcar
+		     (lambda (entry)
+		       (cons (lyrics-fetcher-neteasecloud--format-song-title entry)
+			     (assoc 'id entry)))
+		     results)))
+	       (cdr
+		(assoc
+		 (completing-read
+		  "Pick a result: "
+		  results-songs-for-select
+		  nil t)
+		 results-songs-for-select)))
+	   (assoc 'id (aref results 0))
+	   ))))))
+
+(defun lyrics-fetcher-neteasecloud--format-song-title (entry)
+  "Convert a 'music.163.com' search ENTRY to a string, which can be used in selection."
+  (format "%s by %s"
+	  (cdr (assoc 'name entry))
+	  (cdr (assoc 'name (aref (alist-get 'artists entry) 0)))
+	  ))
+
+(defun lyrics-fetcher-neteasecloud-format-file-name (track)
+  "'Emms' requires lyrics files' name should be the same as their tracks' name except extensions"
```

Comment:
Also, run `M-x checkdoc`. I think it will say that this particular docstring should end with a punctuation and that you should mention `TRACK` somewhere.

Reply:
Thank you for your mention, I have updated docstrings.